src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1689,6 +1689,17 @@ private static bool GetIsWindows7Or2008R2()
             return false;
         }
 
+        internal void HeartBeat()
+        {
+            Http2Connection[]? localHttp2Connections = _http2Connections;
+            if (localHttp2Connections != null)
+            {
+                foreach (Http2Connection http2Connection in localHttp2Connections)
+                    http2Connection?.HeartBeat();
```

Comment:
@alnikola will this collection ever have a null connection in it?

```suggestion
                    http2Connection.HeartBeat();
```

Reply:
Fixed, AFAIK this will never be `null`, the array gets resized with each addition/removal.